var/spack/repos/builtin/packages/py-gdbgui/package.py

```diff
@@ -0,0 +1,43 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class PyGdbgui(PythonPackage):
+    """gdbgui is a modern, free, browser-based frontend to gdb"""
+
+    homepage = "https://gdbgui.com"
+    url      = "https://pypi.io/packages/source/g/gdbgui/gdbgui-0.11.2.1.tar.gz"
+
+    version('0.11.2.1', 'c15dd3f60fe372b2e93f705c7ee75f51')
+
+    depends_on('py-setuptools',             type=('build', 'run'))
```

Comment:
This is only needed at build time, not run time.

Reply:
Actually no, the `gdbgui` binary that is produced in the install process contains an import to `pkg_resources` leading to 
```
sschmitt@hel ➜  ~  gdbgui
Traceback (most recent call last):
  File "/wang/users/wv385/cluster_home/spack_failure/spack2/opt/spack/linux-debian8-x86_64/gcc-4.9.2/py-gdbgui-0.11.2.1-4bqpnhml7wi44uukcvm4cxvpp5yfbjdq/bin/gdbgui", line 7, in <module>
    from pkg_resources import load_entry_point
ImportError: No module named pkg_resources
```
Adding the `setuptools` run dependency fixes that (although I'm not entirely convinced why...)

Reply:
Oh, interesting. I'm surprised that it installs Python scripts that don't exist in the source code. I didn't see any imports to `pkg_resources` in the source, my bad.

> Adding the setuptools run dependency fixes that (although I'm not entirely convinced why...)

`pkg_resources` is one of the modules installed with `py-setuptools`.